lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +174,77 @@ def remove_schedule(name)
 
       private
 
+      def supports_lua?
+        redis_master_version >= 2.5
+      end
+
+      def redis_master_version
+        Resque.redis.info['redis_version'].to_f
+      end
+
+      def startup_sha(refresh = false)
```

Comment:
Yup. From the docs perspective, the user is supposed to set this value before actually "running" the scheduler. Under the hood, what happens is: 
* `@schedule` gets assigned the result of the redis magic that we are trying to patch
* upon running the scheduler (`Resque::Scheduler.run`) before anything happens (i.e. before the polling loop starts) we use `@schedule` to decide what should be scheduled